### PR TITLE
Data hash value

### PR DESCRIPTION
**Pull request by tomvanmele**
### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [ ] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [ ] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/compas/data/data.py [hunk 1]
```diff
@@ -112,12 +113,12 @@ def __init__(self, name=None):
 
     def __getstate__(self):
         """Return the object data for state serialization with older pickle protocols."""
-        return {'__dict__': self.__dict__, 'dtype': self.dtype, 'data': self.data}
+        return {"__dict__": self.__dict__, "dtype": self.dtype, "data": self.data}
```
Comment (gonzalocasas) [inline src/compas/data/data.py:116]: 🖤 black! yay!

File: src/compas/data/data.py [hunk 2]
```diff
@@ -350,3 +360,34 @@ def validate_json(self):
             print("Validation against the JSON schema of this object failed.")
             raise e
         return jsonstring
+
+    def hash(self, as_string=False):
```
Comment (gonzalocasas) [inline src/compas/data/data.py]: This is too close to the built-in `__hash__` magic method, but has different semantics, I would suggest a different name, maybe `sha256hash` or `unique_hash`, or `fingerprint`, or something else

**Review by gonzalocasas — CHANGES_REQUESTED** [review]
the idea looks great to me, I would only change the method name to avoid confusion with the built-in

File: src/compas/data/data.py (hunk 2 above)
Comment (Licini) [reply src/compas/data/data.py]: I vote for `sha256` or `fingerprint`

**Review by gonzalocasas — APPROVED** [review]
LGTM

**Pull request merged**